arch/posix/core/posix_core.c

```diff
@@ -479,9 +479,7 @@ void z_impl_k_thread_abort(k_tid_t thread)
 	unsigned int key;
 	int thread_idx;
 
-	posix_thread_status_t *tstatus =
-					(posix_thread_status_t *)
-					thread->callee_saved.thread_status;
+	posix_thread_status_t *tstatus = thread->callee_saved.thread_status;
```

Comment:
This commit title would be misleading: You are indeed removing here a relatively unnecessary cast ( void * =>posix_thread_status_t * ). But it is only unnecessary after you have changed the type of thread_status.
Could you please change the commit description to mention instead that you are changing the type of a variable which was used as pointer storage from uint32 to void* ?

Reply:
Anyhow, I guess this cast did not really harm there even after being from a (void *), or?